AEPLifecycle/Sources/XDM/XDMLifecycleApplication.swift

```diff
@@ -0,0 +1,59 @@
+//
+// Copyright 2021 Adobe. All rights reserved.
+// This file is licensed to you under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License. You may obtain a copy
+// of the License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under
+// the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+// OF ANY KIND, either express or implied. See the License for the specific language
+// governing permissions and limitations under the License.
+//
+
+import AEPServices
+import Foundation
+
+/// Represents an XDM Lifecycle Application schema
+struct XDMLifecycleApplication {
+    init() {}
+
+    var closeType: XDMCloseType?
+    var id: String?
+    var installDate: Date?
+    var isClose: Bool?
+    var isInstall: Bool?
+    var isLaunch: Bool?
+    var isUpgrade: Bool?
+    var name: String?
+    var sessionLength: Int64?
+    var version: String?
```

Comment:
We should add docs for all these fields as er schema definition - this comment applies for all structs